spec.emu

```diff
@@ -275,27 +273,27 @@ location: https://github.com/tc39/proposal-measure/
             1. If ℝ(_significantDigits_) is not an integer, throw a *RangeError* exception.
             1. If ℝ(_significantDigits_) < 1, throw a *RangeError* exception.
             1. Set _significantDigits_ to ℝ(_significantDigits_).
-        1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Amount.prototype%"*, « [[AmountData]], [[SignificantDigits]], [[FractionDigits]], [[Unit]] »).
+        1. Let _O_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Amount.prototype%"*, « [[Value]], [[SignificantDigits]], [[FractionDigits]], [[Unit]], [[Currency]] »).
         1. Let _parseResult_ be ParseText(_toParse_, |StrDecimalLiteral|).
         1. If _parseResult_ is a List of errors, throw a *SyntaxError* exception.
         1. Let _amountValue_ be ? StringDecimalValue of _parseResult_.
-        1. Let _roundedAmountValue_ be RoundToAmountDomain(_amountValue_, _roundingMode_).
+        1. Let _roundedValue_ be _amountValue_.
         1. If both _significantDigits_ and _fractionDigits_ are *undefined*, then
           1. Set _significantDigits_ to CountSignificantDigits(_toParse_).
           1. Set _fractionDigits_ to CountFractionDigits(_toParse_).
         1. Else if _significantDigits_ is *undefined*, then
-          1. Set _roundedAmountValue_ be RoundToFractionalDigits(_amountValue_, _fractionDigits_, _roundingMode_).
-          1. Let _digitStr_ be the unique decimal string representation of _roundedAmountValue_ without leading zeroes.
+          1. Set _roundedValue_ be RoundToFractionalDigits(_amountValue_, _fractionDigits_, _roundingMode_).
+          1. Let _digitStr_ be the unique decimal string representation of _roundedValue_ without leading zeroes.
           1. Set _significantDigits_ to CountSignificantDigits(_digitStr_).
         1. Otherwise:
-          1. Set _roundedAmountValue_ be RoundToSignificantDigits(_amountValue_, _significantDigits_, _roundingMode_).
-          1. Let _digitStr_ be the unique decimal string representation of _roundedAmountValue_ without leading zeroes.
+          1. Set _roundedValue_ be RoundToSignificantDigits(_amountValue_, _significantDigits_, _roundingMode_).
+          1. Let _digitStr_ be the unique decimal string representation of _roundedValue_ without leading zeroes.
           1. Set _fractionDigits_ to CountFractionDigits(_digitStr_).
-        1. Set _O_.[[AmountData]] to _roundedAmountValue_.
+        1. Set _O_.[[Value]] to _roundedValue_.
         1. Set _O_.[[SignificantDigits]] to _significantDigits_.
         1. Set _O_.[[FractionDigits]] to _fractionDigits_.
         1. If _unit_ is a String, set _O_.[[Unit]] to toLowercase(StringToCodePoints(_unit_)).
-        1. If _currency_ is a String, set _O_.[[Unit]] to toUppercase(StringToCodePoints(_unit_)).
+        1. If _currency_ is a String, set _O_.[[Currency]] to toUppercase(StringToCodePoints(_unit_)).
```

Comment:
We should also validate at least that any string values are not empty, and possibly that they match `/^[\w-]+$/`. For currencies, the validation could even require `/^[A-Z]{3}$/i`.

```suggestion
        1. If _unit_ is a String, set _O_.[[Unit]] to CodePointsToString(toLowercase(StringToCodePoints(_unit_))).
        1. If _currency_ is a String, set _O_.[[Currency]] to CodePointsToString(toUppercase(StringToCodePoints(_currency_))).
```